src/utils/outputs.ts

```diff
@@ -242,3 +242,99 @@ export const getSpeakingNames = (
   });
   return result;
 };
+
+export const getSpeakingNamesWithColors = (
+  speakingNames: SpeakingName[],
+  allColors: Record<string, string>,
+  colors: Record<string, HeisslufType[]>,
+  darkMode: boolean,
+  luminanceSteps: number[],
+  speakingNamesDefaultMapping: SpeakingName[]
+): any => {
+  let result: any = {};
+  let state;
+  const isStateName = (name: string) => {
+    return (
+      name.includes("enabled") ||
+      name.includes("hover") ||
+      name.includes("pressed")
+    );
+  };
+
+  const processState = (name: string) => {
+    const stateIndex = name.lastIndexOf("-");
+    state = name.slice(stateIndex + 1);
+    const nameWithoutState = name.slice(0, stateIndex);
+
+    const numOfState =
+      state === "enabled" ? "01" : state === "hover" ? "02" : "03";
+    const stateNumbered = `${numOfState}-${state}`;
+
+    return [nameWithoutState, numOfState, state, stateNumbered];
+  };
+
+  Object.entries(colors).forEach(([name, color]) => {
+    if (name === "brand") {
+      const brandColor = allColors["brand"];
+      const lightBrand = getExtraBrandColors(brandColor, false, luminanceSteps);
+      const darkBrand = getExtraBrandColors(brandColor, true, luminanceSteps);
+
+      const brandTheme = darkMode ? darkBrand : lightBrand;
+
+      result = {
+        ...result,
+        "brand/On/01-Enabled": `transparency 0%, ${brandTheme.brandOnColor}`,
+        "brand/Origin/01-Enabled": `transparency 0%, ${brandTheme.color}`,
+        "brand/Origin/02-Hover": `transparency 0%, ${brandTheme.hoverColor}`,
+        "brand/Origin/03-Pressed": `transparency 0%, ${brandTheme.pressedColor}`,
+      };
+    }
+
+    speakingNames.forEach((speakingName) => {
+      const mappedName = speakingNamesDefaultMapping.find(
+        (mapping) => mapping.name === speakingName.name
+      );
+
+      if (mappedName) {
+        const colorNumber = darkMode ? mappedName.dark : mappedName.light;
+        const hexValue = color[colorNumber].hex;
+
+        let transparency;
+        if (
+          speakingName.transparencyDark !== undefined ||
+          speakingName.transparencyLight !== undefined
+        ) {
+          transparency = darkMode
+            ? speakingName.transparencyDark
+            : speakingName.transparencyLight;
+        } else {
+          transparency = 0;
+        }
+
+        if (speakingName.name.includes("on-")) {
+          const nameWithoutOnPrefix = speakingName.name.replace("on-", "");
+          if (isStateName(speakingName.name)) {
+            const processedState = processState(nameWithoutOnPrefix);
```

Comment:
If `processState` would return an object you could do something like:

`const {nameWithoutState, numOfState, state, stateNumbered} = processState(nameWithoutOnPrefix);`

So you could access all props directly without using an index on the array